internal/vds/vds_test.go

```diff
@@ -315,3 +315,83 @@ func TestSliceMetadataAxisOrdering(t *testing.T) {
 		}
 	}
 }
+
+func TestFence(t *testing.T) {
+	expected := []float32{
+		108, 109, 110, 111, // il: 3, xl: 10, samples: all
+		112, 113, 114, 115, // il: 3, xl: 11, samples: all
+		100, 101, 102, 103, // il: 1, xl: 10, samples: all
+        108, 109, 110, 111, // il: 3, xl: 10, samples: all
+        116, 117, 118, 119, // il: 5, xl: 10, samples: all
+	}
+
+	testcases := []struct{
+		coordinate_system string
+		coordinates       [][]float32
+	} {
+		{
+			coordinate_system: "ij",
+			coordinates:
+				[][]float32{{1, 0}, {1, 1}, {0, 0}, {1, 0}, {2, 0}},
+		},
+		{
+			coordinate_system: "ilxl",
+			coordinates:
+				[][]float32{{3, 10}, {3, 11}, {1, 10}, {3, 10}, {5, 10}},
+		},
+		{
+			coordinate_system: "cdp",
+			coordinates:
+				[][]float32{{3.2, 3}, {3.2, 6.3}, {1, 3}, {3.2, 3}, {5.4, 3}},
```

Comment:
The coordinates are passed directly to openvds, so you can expect whatever behavior that provides. We are using "nearest" interpolation method, so it should in fact be the nearest. 

Not sure if it makes sense to add many tests that are actually just testing openvds (these tests should rather be contributed directly to openvds).

Reply:
That I would argue falls under "blacbox test important for user". Tests here are supposed to prove 1) we use library correctly 2) library itself can be trusted - because we do not trust it enough yet

1. If we internally one day change interpolation method from nearest to something else which changes behavior for user, it would be nice to know.
2. I, as a developer of this system, want to have an idea that openvds (and thus our code) returns me something totally pointless. Because in my opinion something definitely should be done with it. Because I would not define that as expected behavior of our system.

Interesting situation here is that if I change current interpolation method, our current "point exists" tests fail. Makes me wonder if this is due to interpolation on artificial file, or if this has something to do with our "race" problem.

Reply:
We will most likely soon expose different interpolation methods through parameters. The way I see it we don't support interpolation yet, it just happens to work because upenvds supports it. So since this it as of now incidental behaviour, I suggest we don't test it yet.